presto-native-execution/presto_cpp/main/SessionProperties.cpp

```diff
@@ -0,0 +1,261 @@
+
+  addSessionProperty(
+      kWriterSpillEnabled,
+      "Native Execution only. Enable writer spilling on native engine",
+      BOOLEAN(),
+      false,
+      QueryConfig::kWriterSpillEnabled,
+      boolToString(c.writerSpillEnabled()));
+
+  addSessionProperty(
+      kRowNumberSpillEnabled,
+      "Native Execution only. Enable row number spilling on native engine",
+      BOOLEAN(),
+      false,
+      QueryConfig::kRowNumberSpillEnabled,
+      boolToString(c.rowNumberSpillEnabled()));
+
+  addSessionProperty(
+      kJoinSpillPartitionBits,
+      "Native Execution only. The number of bits (N) used to calculate the "
+      "spilling partition number for hash join and RowNumber: 2 ^ N",
+      INTEGER(),
+      false,
+      QueryConfig::kJoinSpillPartitionBits,
+      std::to_string(c.rowNumberSpillEnabled()));
+
+  addSessionProperty(
+      kNativeSpillerNumPartitionBits,
+      "none",
+      TINYINT(),
+      false,
+      QueryConfig::kSpillNumPartitionBits,
+      std::to_string(c.spillNumPartitionBits())),
+
+      addSessionProperty(
+          kTopNRowNumberSpillEnabled,
+          "Native Execution only. Enable topN row number spilling on native engine",
+          BOOLEAN(),
+          false,
+          QueryConfig::kTopNRowNumberSpillEnabled,
+          boolToString(c.topNRowNumberSpillEnabled()));
+
+  addSessionProperty(
+      kValidateOutputFromOperators,
+      "If set to true, then during execution of tasks, the output vectors of "
+      "every operator are validated for consistency. This is an expensive check "
+      "so should only be used for debugging. It can help debug issues where "
+      "malformed vector cause failures or crashes by helping identify which "
+      "operator is generating them.",
+      BOOLEAN(),
+      false,
+      QueryConfig::kValidateOutputFromOperators,
+      boolToString(c.validateOutputFromOperators()));
+
+  // If `legacy_timestamp` is true, the coordinator expects timestamp
+  // conversions without a timezone to be converted to the user's
+  // session_timezone.
+  addSessionProperty(
+      kLegacyTimestamp,
+      "Native Execution only. Use legacy TIME & TIMESTAMP semantics. Warning: "
+      "this will be removed",
+      BOOLEAN(),
+      false,
+      QueryConfig::kAdjustTimestampToTimezone,
+      // Overrides velox default value. legacy_timestamp default value is true
+      // in the coordinator.
+      "true");
+
+  // TODO: remove this once cpu driver slicing config is turned on by default in
+  // Velox.
+  addSessionProperty(
+      kDriverCpuTimeSliceLimitMs,
+      "Native Execution only. The cpu time slice limit in ms that a driver thread. "
+      "If not zero, can continuously run without yielding. If it is zero, then "
+      "there is no limit.",
+      INTEGER(),
+      false,
+      QueryConfig::kDriverCpuTimeSliceLimitMs,
+      // Overrides velox default value. Set it to 1 second to be aligned with
+      // Presto Java.
```

Comment:
Two hard coded properties were added by updateVeloxConfigs() are already included and these properties will be visible in show session. Please remove updateVeloxConfigs() from QueryContextManager::findOrCreateQueryCtx().